### PR TITLE
939 document vars populated by derive param

### PR DESCRIPTION
**Pull request by sophie-gem**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: inst/templates/ad_adex.R [hunk 1]
```diff
@@ -95,7 +95,9 @@ adex <- bind_rows(
   mutate(PARCAT1 = "INDIVIDUAL")
 
   # Part 3
-  # Derive summary parameters
+  # Derive summary parameters. Note that, for the functions `derive_param_exposure`,
+  # `derive_param_doseint` and `derive_derived_param`, only the variables specified
+  # in `by_vars` will be populated in the newly created records.
```
Comment (thomas-neitmann) [inline inst/templates/ad_adex.R:100]: Please add a set of trailing parentheses after the function names, e.g. `derive_param_doseint()`.

File: inst/templates/ad_adeg.R [hunk 2]
```diff
@@ -125,7 +125,9 @@ adeg <- adeg %>%
     AVALC = EGSTRESC
   ) %>%
 
-  # Derive new parameters based on existing records.
+  # Derive new parameters based on existing records. Note that, for the following
+  # four derive_param_* functions, only the variables specified in `by_vars` will
+  # be populated in the newly created records.
```
Comment (thomas-neitmann) [inline inst/templates/ad_adeg.R:130]: Please put `derive_param_*()` in backticks add add a set of trailing parentheses.

File: inst/templates/ad_advs.R [hunk 3]
```diff
@@ -107,7 +107,9 @@ advs <- advs %>%
     AVALC = VSSTRESC
   ) %>%
 
-  # Derive new parameters based on existing records.
+  # Derive new parameters based on existing records. Note that, for the following
+  # three derive_param_* functions, only the variables specified in `by_vars` will
+  # be populated in the newly created records.
```
Comment (thomas-neitmann) [inline inst/templates/ad_advs.R:112]: Please put `derive_param_*()` in backticks add add a set of trailing parentheses.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]
Thanks a lot @sophie-gem! Just a minor comment. Please fix and then we can merge 🙂

File: inst/templates/ad_advs.R (hunk 3 above)
Comment (sophie-gem) [reply inst/templates/ad_advs.R:112]: Yes of course, I will do this now. Is this just in the template programs or shall I check the vignettes also?

**Comment by sophie-gem** [comment]
Hi @thomas-neitmann , believe the most recent commit should have fixed the requests above. Regarding running `pkgdown::build_site()` and `devtools::check()` - these both come back with errors saying that `admiraltest` package doesn't exist. Let me know if there is anything else required here.

**Comment by sophie-gem** [comment]
@thomas-neitmann - do you happen to know where `dose_freq_lookup` is defined? As this is causing the GH Action checks to fail?

**Comment by bundfussr** [comment]
`dose_freq_lookup` is defined in `create_single_dose_dataset.R`.

**Comment by sophie-gem** [comment]
> `dose_freq_lookup` is defined in `create_single_dose_dataset.R`.

Great thanks @bundfussr , will look into that now - don't know why this is failing as I haven't touched that script!

**Comment by sophie-gem** [comment]
@thomas-neitmann , @bundfussr I have found what the issue is with the `dose_freq_lookup` documentation (which I believe is what is causing the checks to fail). In the description there is a markdown link containing a `%` sign:
```
#' These pre-defined dose frequencies are sourced from
#' [CDISC](https://evs.nci.nih.gov/ftp1/CDISC/SDTM/SDTM%20Terminology.pdf). The
#' number of rows to generate using `create_single_dose_dataset()` arguments
#' `start_date` and `end_date` is derived from `DOSE_COUNT`, `DOSE_WINDOW`, and
#' `CONVERSION_FACTOR` with appropriate functions from `lubridate`.
```
As explained [here](https://roxygen2.r-lib.org/news/index.html#roxygen2-712) - this was an issue with `roxygen2` fixed only in version 7.1.2 (I believe `admiral` is currently using v7.0.2 - but let me know if I'm wrong.)

**Comment by bundfussr** [comment]
@sophie-gem , @thomas-neitmann , once we were using version 7.1.2. But renv.lock was updated to use older versions (of roxygen2 and other packages). I assume this was not intended. However, I am not sure what the correct versions are.

**Comment by bms63** [comment]
Hi @sophie-gem.  Are you able to finish this today?  Any blockers you are facing?  Seems like you are very close.  Thanks again for all your hard work!!

**Comment by sophie-gem** [comment]
Hi @bms63, of course - I'll get onto it after work today. Hopefully shouldn't take too long.

**Review by thomas-neitmann — APPROVED** [review]

**Comment by bms63** [comment]
Looks like the linter check is falling over for this one

**Pull request merged**